src/settings.h

```diff
@@ -217,6 +218,35 @@ class Settings
         ms_registryPath = path;
     }
 
+    /// Set custom configuration read, write and delete functions
+    static void SetConfigMethods(void *customConfigData, 
+                                win_sparkle_config_read_t customConfigRead,
+                                win_sparkle_config_write_t customConfigWrite, 
+                                win_sparkle_config_delete_t customConfigDelete)
+    {
+        CriticalSectionLocker lock(ms_csVars);
+        
+        ms_currentConfigRead = DefaultConfigRead;
+        ms_currentConfigWrite = DefaultConfigWrite;
+        ms_currentConfigDelete = DefaultConfigDelete;
```

Comment:
So now there is a small amount of time between this line and the ones below, when `ms_currentConfig*` may be `nullptr`.
To avoid the most unreal scenario, when one of `Do*ConfigValue` functions called in exactly this moment, I suggest either lock `g_csConfigValues` in this function as well, either move these assignments under `else` below.